cdap-ui/app/cdap/services/helpers.js

```diff
@@ -126,8 +126,9 @@ function isDescendant(parent, child) {
 function getArtifactNameAndVersion (nameWithVersion) {
   // core-plugins-3.4.0-SNAPSHOT.jar
   // extracts version from the jar file name. We then get the name of the artifact (that is from the beginning up to version beginning)
-  let regExpRule = new RegExp('(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?$');
-  let version = regExpRule.exec(nameWithVersion)[0];
+  // Fixed it to use a suffix pattern. Added `\\-` to detect versions from names such as `redshifttos3-action-plugin-1.0.0.json`
+  let regExpRule = new RegExp('\\-(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?$');
+  let version = regExpRule.exec(nameWithVersion)[0].slice(1);
```

Comment:
So now does `regExpRule.exec(nameWithVersion)[0]` return an array instead of just a string?